src/Umbraco.Web/Routing/UrlProviderExtensions.cs

```diff
@@ -1,117 +1,175 @@
+            var uri = new Uri(url.TrimEnd('/'), UriKind.RelativeOrAbsolute);
+            if (uri.IsAbsoluteUri == false) uri = uri.MakeAbsolute(UmbracoContext.Current.CleanedUmbracoUrl);
+            uri = UriUtility.UriToUmbraco(uri);
+            var pcr = publishedRouter.CreateRequest(UmbracoContext.Current, uri);
+            publishedRouter.TryRouteRequest(pcr);
+
+            if (pcr.HasPublishedContent == false)
             {
-                urls.Add(textService.Localize("content/getUrlException"));
+                urls.Add(UrlInfo.Message(textService.Localize("content/routeErrorCannotRoute"), culture));
+                return true;
             }
-            else
+
+            if (pcr.PublishedContent.Id != content.Id)
             {
-                // test for collisions
-                var uri = new Uri(url.TrimEnd('/'), UriKind.RelativeOrAbsolute);
-                if (uri.IsAbsoluteUri == false) uri = uri.MakeAbsolute(UmbracoContext.Current.CleanedUmbracoUrl);
-                uri = UriUtility.UriToUmbraco(uri);
-                var r = Core.Composing.Current.Container.GetInstance<PublishedRouter>(); // fixme inject or ?
-                var pcr = r.CreateRequest(UmbracoContext.Current, uri);
-                r.TryRouteRequest(pcr);
-
-                if (pcr.HasPublishedContent == false)
-                {
-                    urls.Add(textService.Localize("content/routeError", new[] { "(error)" }));
-                }
-                else if (pcr.PublishedContent.Id != content.Id)
+                var o = pcr.PublishedContent;
+                var l = new List<string>();
+                while (o != null)
                 {
-                    var o = pcr.PublishedContent;
-                    string s;
-                    if (o == null)
-                    {
-                        s = "(unknown)";
-                    }
-                    else
-                    {
-                        var l = new List<string>();
-                        while (o != null)
-                        {
-                            l.Add(o.Name);
-                            o = o.Parent;
-                        }
-                        l.Reverse();
-                        s = "/" + string.Join("/", l) + " (id=" + pcr.PublishedContent.Id + ")";
-
-                    }
-                    urls.Add(textService.Localize("content/routeError", s));
-                }
-                else
-                {
-                    urls.Add(url);
-                    foreach(var otherUrl in urlProvider.GetOtherUrls(content.Id))
-                    {
-                        urls.Add(otherUrl);
-                    }
+                    l.Add(o.Name);
+                    o = o.Parent;
                 }
+                l.Reverse();
+                var s = "/" + string.Join("/", l) + " (id=" + pcr.PublishedContent.Id + ")";
+
+                urls.Add(UrlInfo.Message(textService.Localize("content/routeError", s), culture));
```

Comment:
To pass in tokens it needs to be an array like `new[]{ s }`, i have fixed in rev 7250b3d49198e908facde316d61f7fb8d0a0835f